datetime/src/Data/DateTime.ss

```diff
@@ -0,0 +1,74 @@
+;; -*- mode: scheme -*-
+
+(library (Data.DateTime foreign)
+  (export calcDiff
+          adjustImpl)
+  (import (only (rnrs base) define lambda let let*)
+          (prefix (purs runtime) rt:)
+          (prefix (chezscheme) scm:))
+
+  ;; Note: make-date in chez has a lower bound at 1901:
+  ;; https://github.com/cisco/ChezScheme/blob/57f92bb76aed694437a2e201780654e9c03a576f/s/date.ss#L186
+  (define make-date
+    (lambda (millisecond second minute hour day month year)
+      (scm:make-date (scm:fx* 1000000 millisecond) second minute hour day month year 0)))
+
+  (define calcDiff
+    (lambda (datetimeRecord1 datetimeRecord2)
+      (let* ([year1 (rt:record-ref datetimeRecord1 (scm:string->symbol "year"))]
+            [month1 (rt:record-ref datetimeRecord1 (scm:string->symbol "month"))]
+            [day1 (rt:record-ref datetimeRecord1 (scm:string->symbol "day"))]
+            [hour1 (rt:record-ref datetimeRecord1 (scm:string->symbol "hour"))]
+            [minute1 (rt:record-ref datetimeRecord1 (scm:string->symbol "minute"))]
+            [second1 (rt:record-ref datetimeRecord1 (scm:string->symbol "second"))]
+            [millisecond1 (rt:record-ref datetimeRecord1 (scm:string->symbol "millisecond"))]
+            [year2 (rt:record-ref datetimeRecord2 (scm:string->symbol "year"))]
+            [month2 (rt:record-ref datetimeRecord2 (scm:string->symbol "month"))]
+            [day2 (rt:record-ref datetimeRecord2 (scm:string->symbol "day"))]
+            [hour2 (rt:record-ref datetimeRecord2 (scm:string->symbol "hour"))]
+            [minute2 (rt:record-ref datetimeRecord2 (scm:string->symbol "minute"))]
+            [second2 (rt:record-ref datetimeRecord2 (scm:string->symbol "second"))]
+            [millisecond2 (rt:record-ref datetimeRecord2 (scm:string->symbol "millisecond"))]
```

Comment:
Why not use symbol literals here and below?

Reply:
Like `(scm:quote millisecond)`?

Reply:
Done